### PR TITLE
Migrate sln files when running dotnet migrate in a folder

### PR DESCRIPTION
**Pull request by livarcocc**
Fixing our solution tests so that they go back to building the solution. Seems to work now, at least on OSX.
Also, made a change to update solution to update any solutions in the folder where dotnet migrate is executed.
Also, made a change to find dependencies when migrating in a folder recursively.

Fixes https://github.com/dotnet/cli/issues/5238

@piotrpMSFT @krwq @jgoshi @jonsequitur

### CODE REVIEW

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateSolutions.cs [hunk 1]
```diff
@@ -204,11 +256,10 @@ private void MigrateAndBuild(string groupName, string projectName, [CallerMember
                 .Execute($"restore \"{solutionRelPath}\"")
                 .Should().Pass();
 
-            //ISSUE: https://github.com/dotnet/cli/issues/5205
```
Comment (jgoshi) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateSolutions.cs]: This appears to still fail (sporadically) on other platforms (looking at your test failures).

Comment (jgoshi) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateSolutions.cs]: If it does actually work (you re-enable it), then there is another test to re-enable that is mentioned in one of the issues you reference.

Comment (livarcocc) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateSolutions.cs]: You are right. This still reproes. Commented the build step out again.

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x86 Debug Build

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please.

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x86 Debug Build

**Pull request merged**